src/components/display-card/CardDeck.js

```diff
@@ -1,21 +1,27 @@
 /* eslint-disable no-unused-vars */
 import * as React from 'react';
-import { Box, Button, Card, CardContent, Typography, CardActions } from "@material-ui/core";
-import Logo from "../game-intro/resources/logo.svg";
-import { useLocation } from 'react-router-dom';
+import { Button } from "@material-ui/core";
+import {useLocation, useNavigate} from 'react-router-dom';
 import DisplayCard from './DisplayCard';
+import csvHelper from "./helper/csv_helper";
+import CSVDownload from "react-csv/src/components/Download";
+
 
 function CardDeck() {
 
     const location = useLocation()
     const cards = location.state.cards
-    
+    const navigate = useNavigate()
+
     const [currentCard, setCurrentCard] = React.useState([])
     const [cardLength, setCardLength] = React.useState(cards?.length)
     const [sentences, setSentences] = React.useState(location.state.sentences)
     const [choice, setChoice] = React.useState("");
     const [score, setScore] = React.useState({player: 0, robot: 0, enable: false})
     const [order, setOrder] = React.useState([0, 1, 2])
+    const [csvRecord, setCsvRecord] = React.useState([])
+    const [visitedCards, setVisitedCards] = React.useState([])
+    const [visitecChoices, setVisitecChoices] = React.useState([])
```

Comment:
Can we try to use just one const instead of two so that we don't need to maintain a lot of variables?

Reply:
done